src/test/java/core/basesyntax/service/impl/SubtractOperationHandlerImplTest.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SubtractOperationHandlerImplTest {
+    private static final String KEY_APPLE = "apple";
+
+    @BeforeClass
+    public static void beforeClass() {
+        FruitDao fruitDao = new FruitDaoImpl();
+        Map<FruitTransaction.Operation, OperationHandler> handlerMap = new HashMap<>();
+        handlerMap.put(FruitTransaction.Operation.SUPPLY, new AddOperationHandlerImpl(fruitDao));
+        handlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new SubtractOperationHandlerImpl(fruitDao));
+        List<FruitTransaction> list = new ArrayList<>();
+        list.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,KEY_APPLE,10));
+        list.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,KEY_APPLE,10));
+        AddOperationHandlerImpl addOperationHandler = new AddOperationHandlerImpl(fruitDao);
+        addOperationHandler.operationWithFruitTransaction(list.get(0));
+        SubtractOperationHandlerImpl subtractOperationHandler =
+                new SubtractOperationHandlerImpl(fruitDao);
+        subtractOperationHandler.operationWithFruitTransaction(list.get(1));
+    }
```

Comment:
`beforeClass` should initialise fields that we need to run tests, like `SubtractOperationHandlerImpl`
but variables like `list` and some logic like `subtractOperationHandler.operationWithFruitTransaction(list.get(1));`
should be inside a test method
please, check it everywhere